### ISSUE TITLE
AttributeError: 'module' object has no attribute 'mysql'

### ISSUE THREAD

**Issue by qdgtruie** [Murodese/pynab#194]
I did a git pull and now I get 

Traceback (most recent call last):
  File "pynab.py", line 23, in <module>
    import pynab.util
  File "/usr/local/www/pynab-postgres/pynab/__init__.py", line 107, in <module>
    check_config()
  File "/usr/local/www/pynab-postgres/pynab/__init__.py", line 34, in check_config
    inner_level = set(getattr(config_sample, item).keys()) - set(getattr(config, item).keys())
AttributeError: 'module' object has no attribute 'mysql'

I thought I was missing something on the conf but it is not the case...

**Comment by qdgtruie** [comment]
ok I fixed it ...

**Comment by Murodese** [comment]
Works now?

**Comment by qdgtruie** [comment]
this part yes.
but it seems that there are some other strong changes (especially the supervisor dependency)

root@pynab:/usr/local/www/pynab-postgres # python3.3 pynab.py start
pynab:scan: ERROR (no such process)
pynab:postproc: ERROR (no such process)
pynab:prebot: ERROR (no such process)
pynab:stats: ERROR (no such process)

not sure what is happening

**Comment by Murodese** [comment]
What did you update from/to? Did you update using `git pull` or `./pynab.py update`?

**Comment by qdgtruie** [comment]
git pull

but I eventually found out .. the supervisor thing is used to bootstrap the program now and the supervisor configuration (including the init/supervisor/pynab.conf) makes lot of assumptions about the system and filesystem layout (I am not runing linux but BSD)

**Comment by qdgtruie** [comment]
now it seems it works

**Comment by Murodese** [comment]
Supervisor should work on any *nix OS, but yeah, you'll have to modify some of the configs to suit your paths etc. Glad to hear it works!

**Comment by qdgtruie** [comment]
some remarks on supervisor : 
- path to python is assuming linux (vs /usr/local/bin on BSD)
- path to pynab is assuming /opt (instead of /usr/local/www convention for BSD)
- username for the processs is assuming the user "www-data" does exist
- /etc/supervisor/conf.d/ does not exist on some systems
so basically a large part of the conf is to be rewritten (in my case)

**Comment by qdgtruie** [comment]
I tried to use the  "python3.3 pynab.py update" command and it is stuck on 
INFO  [alembic.migration] Running upgrade 1de31e1a78b -> 3379cecd3a7, un-black all rars and nfos

how to check what is happening ?

**Comment by Murodese** [comment]
It's not stuck. Some of the migrations do some fairly heavy-duty query work, particularly if there's a lot of releases in your db, so it'll take some time.

**Comment by qdgtruie** [comment]
it has been running for 12h and the load averages are :  0.00,  0.00,  0.00  

but it is still stuck on the same step "INFO [alembic.migration] Running upgrade 1de31e1a78b -> 3379cecd3a7, un-black all rars and nfos"

**Comment by qdgtruie** [comment]
now it has running for 2 days still no visible activity and still on the same step

 "INFO [alembic.migration] Running upgrade 1de31e1a78b -> 3379cecd3a7, un-black all rars and nfos"

**Comment by Murodese** [comment]
Most likely it's deadlocked. Did you have any of the scripts running while it was upgrading?

Try killing any active scripts, restarting the postgres server and re-running the upgrade.

**Comment by qdgtruie** [comment]
I did it and it ran into deadlock again .
I connected to postgres to check the granted locks and this is what I get : 

UPDATE alembic_version SET version_num='1de31e1a78b' WHERE alembic_version.version_num = '91a9b24e98'

is blocking the following statement

SELECT releases.id AS releases_id FROM releases, metablack  WHERE releases.rar_metablack_id = metablack.id AND metablack.status = 'IMPOSSIBLE'

I am not sure why a simple update on version_num should lock a totaly separate statement (table lock ?)
 
any idea ?

**Comment by Murodese** [comment]
I think that's probably just postgres being weird. Luckily, that particular migration's not that important. If you restart postgres, you can disable that migration by going into `alembic/versions/3379...` and commenting out the upgrade block.

**Comment by bettse** [comment]
I ran into the same original error (`AttributeError: 'module' object has no attribute 'mysql'`), and I found that I still had a mysql section in my config file.  I think that the logic for checking the configs doesn't work as well when the personalized config has a section that the sample doesn't.  When the sample has a new section of field, I would see the 'missing ...' stuff, but I think the opposite isn't true.
